### ISSUE TITLE
Follow strict semantic versioning in package.json

### ISSUE THREAD

**Issue by joelpurra** [getdnsapi/getdns-node#15]
As specified in [semantic versioning](http://semver.org/) v2.0.0, [pre-release version numbers](http://semver.org/#spec-item-9) must have a hyphen to separate the patch version and the pre-release name: `v1.0.0-b2`.

> [9.](http://semver.org/#spec-item-9) A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphen [0-9A-Za-z-]. Identifiers MUST NOT be empty. Numeric identifiers MUST NOT include leading zeroes. Pre-release versions have a lower precedence than the associated normal version. A pre-release version indicates that the version is unstable and might not satisfy the intended compatibility requirements as denoted by its associated normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92.

The current getdns pre-release naming is in the style `v1.0.0b2` and `v1.1.0a1`. The current getdns-node `package.json` has copied this format. The format is already good and useful, but in the spirit of following the semver standard, I propose that future pre-releases of getdns-node, independently of the upstream getdns, introduce this hyphen, as the default package manager for [node.js](https://nodejs.org), [`npm`](https://npmjs.com/) heavily [relies on semver](https://docs.npmjs.com/files/package.json#version) for [version matching](https://docs.npmjs.com/misc/semver). The version number formatting impact is higher for node-based applications and will most likely be updated to be on the safe side when it comes to release management.

**Comment by joelpurra** [comment]
The published version of [`getdns-node` in NPM already has `v1.0.0`](https://www.npmjs.com/package/getdns). This version number does not match any number in `package.json` in the git repository, as far as I can tell. For this reason I have tagged an arbitrary (but quite recent) commit as v1.0.0. The fact that it targets the upstream getdns v1.0.0 is incidental -- it was actually targeting a prerelease at the time of publishing.

After digging around, I found a breaking mismatch between the code/tests and the documentation. The documentation seems to be right, but due to breaking code compatibility from 2014, it's best to bump the major version. This makes the next release of getdns-node v2.0.0, targeting getdns v1.0.0. (Or perhaps getdns v1.1.0, but should publish at least once in between.)

Breaking away from staying in sync with the upstream getdns version would be liberating for getdns-node. It would also allow, for example, moving towards full promise support before getdns reaches v2.0.0.

**Comment by joelpurra** [comment]
Also, if supporting older versions of the upstream getdns is/becomes important, I suggest using [git-flow](http://danielkummer.github.io/git-flow-cheatsheet/) ([`git flow support`](https://github.com/petervanderdoes/gitflow-avh)) `support/getdns-v0.9.x` style branches and a matching `getdns-v0.9.x` as a unique/publishable NPM package name. This should allow backporting of getdns patches affecting the getdns-code as well as version-dependent hotfixes to getdns-node, while keeping both git clean(ish) and NPM packages separately targetable.

**Comment by joelpurra** [comment]
getdns-node is now using  [git-flow](http://danielkummer.github.io/git-flow-cheatsheet/) (preferably with the [`git flow` tooling](https://github.com/petervanderdoes/gitflow-avh)).

To enable in your local clone:

```shell
git flow init -d
```

Expected/sample output:

```text
Using default branch names.

Which branch should be used for bringing forth production releases?
   - develop
   - master
Branch name for production releases: [master]

Which branch should be used for integration of the "next release"?
   - develop
Branch name for "next release" development: [develop]

How to name your supporting branch prefixes?
Feature branches? [feature/]
Bugfix branches? [bugfix/]
Release branches? [release/]
Hotfix branches? [hotfix/]
Support branches? [support/]
Version tag prefix? []
Hooks and filters directory? [$HOME/code/getdns-node/.git/hooks]
```

This should be [mentioned in the readme](https://github.com/getdnsapi/getdns-node#building-and-testing).

**Comment by joelpurra** [comment]
There have been many changes since [v1.0.0](https://github.com/getdnsapi/getdns-node/tree/v1.0.0) (which has an [arbitrary commit](https://github.com/getdnsapi/getdns-node/issues/15#issuecomment-289222381)). For this reason the next release on npm will for sure be v2.0.0.

**Comment by joelpurra** [comment]
With getdns-node v2.0.0-alpha.1 being released (see #22) I'll close this issue.
